app/services/schedule_data_manager_service.rb

```diff
@@ -53,31 +88,46 @@ def temperatures
       end
       data
     end
-  end
 
-  def solar_pv
-    cache_key = "#{@solar_pv_tuos_area_id}-solar-pv-2-tuos"
-    @solar_pv ||= Rails.cache.fetch(cache_key, expires_in: 3.hours) do
-      data = SolarPV.new('solar pv')
-      DataFeeds::SolarPvTuosReading.where(area_id: @solar_pv_tuos_area_id).pluck(:reading_date, :generation_mw_x48).each do |date, values|
-        data.add(date, values.map(&:to_f))
-      end
-      data
-    end
+    # Only use temperature data within datetime bounds of school meter readings
+    return cached_temperatures unless school_reading_date_bounds.present?
+
+    dates_to_remove = cached_temperatures.keys.select { |date| !date.between?(*school_reading_date_bounds) }
+    cached_temperatures.remove_dates!(*dates_to_remove)
   end
 
-  def uk_grid_carbon_intensity
-    cache_key = "co2-feed"
-    @uk_grid_carbon_intensity ||= Rails.cache.fetch(cache_key, expires_in: 3.hours) do
-      uk_grid_carbon_intensity_data = GridCarbonIntensity.new
-      DataFeeds::CarbonIntensityReading.all.pluck(:reading_date, :carbon_intensity_x48).each do |date, values|
-        uk_grid_carbon_intensity_data.add(date, values.map(&:to_f))
+  def find_holidays
+    cached_holidays ||= Rails.cache.fetch(self.class.calendar_cache_key(@calendar), expires_in: 3.hours) do
+      hol_data = HolidayData.new
+
+      Calendar.find(@calendar.id).outside_term_time.order(:start_date).includes(:academic_year, :calendar_event_type).map do |holiday|
+        academic_year = nil # Not really being used at the moment by the analytics code
+
+        analytics_holiday = Holiday.new(
+          holiday.calendar_event_type.analytics_event_type.to_sym,
+          nil,
+          holiday.start_date,
+          holiday.end_date,
+          academic_year
+        )
+
+        hol_data.add(analytics_holiday)
       end
-      uk_grid_carbon_intensity_data
+      Holidays.new(hol_data)
     end
+
+    # Only use holiday data within datetime bounds of school meter readings
```

Comment:
Suggest we leave all holiday data for the moment. Its a relatively small amount of data compared to temperature, solar, etc.

Reply:
:+1: